pkg/hostedMasterApi/v20170831/types.go

```diff
@@ -0,0 +1,224 @@
+package v20170831
+
+import "encoding/json"
+
+// The validate tag is used for validation
+// Reference to gopkg.in/go-playground/validator.v9
+
+// ResourcePurchasePlan defines resource plan as required by ARM
+// for billing purposes.
+type ResourcePurchasePlan struct {
+	Name          string `json:"name,omitempty"`
+	Product       string `json:"product,omitempty"`
+	PromotionCode string `json:"promotionCode,omitempty"`
+	Publisher     string `json:"publisher,omitempty"`
+}
+
+// ContainerService complies with the ARM model of
+// resource definition in a JSON template.
+type ContainerService struct {
+	ID       string                `json:"id,omitempty"`
+	Location string                `json:"location,omitempty" validate:"required"`
+	Name     string                `json:"name,omitempty"`
+	Plan     *ResourcePurchasePlan `json:"plan,omitempty"`
+	Tags     map[string]string     `json:"tags,omitempty"`
+	Type     string                `json:"type,omitempty"`
+
+	Properties *Properties `json:"properties"`
+}
+
+// Properties represents the ACS cluster definition
+type Properties struct {
+	ProvisioningState ProvisioningState   `json:"provisioningState,omitempty"`
+	Version           OrchestratorVersion `json:"version"`
```

Comment:
Isn't calling it KubernetesVersion redundant (like it's not prepending Kubernetes to other fields)? Plus I thought this was signed off by all in the JSON of the API that I shared last week.